velox/functions/lib/Re2Functions.cpp

```diff
@@ -125,18 +125,21 @@ bool re2Extract(
   }
 }
 
-std::string
-likePatternToRe2(StringView pattern, char escapeChar, bool& validPattern) {
+std::string likePatternToRe2(
+    StringView pattern,
+    bool hasEscape,
```

Comment:
nit: using std::optional for escapeChar might be clearer

Reply:
You sure? The code becomes a lot cleaner elsewhere (see other comments).